### PR TITLE
Karma tests fix

### PR DESCRIPTION
**Pull request by maniax89**
Tests weren't passing without the shim and some additional fixes.
#3

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -1,8 +1,7 @@
...
 
 module.exports = env=> {
-    return [
-        require('./config/webpack-dev.config'),
-        require('./config/webpack-prod.config')
-    ];
-};
\ No newline at end of file
+  return [
+    require(`./config/webpack-${env}.config`),
```
Comment (pavanpodila) [inline webpack.config.js:5]: This only includes one of the configs, where previously we had both dev + prod. Will have to check if the demo works fine?

File: config/webpack-prod.config.js [hunk 2]
```diff
@@ -11,9 +11,7 @@ module.exports = merge(webpackCommon, {
     },
     output: {
         filename: '[name].js',
-        path: path.resolve(__dirname, '../dist'),
-        library: true,
```
Comment (pavanpodila) [inline config/webpack-prod.config.js:15]: Any reason why we are removing this. Consumers will not be able to use this directly as a lib. Also need to check if the webpackbin demo continues to work with these changes

**Review by pavanpodila — COMMENTED** [review]
Need some clarification

File: webpack.config.js (hunk 1 above)
Comment (maniax89) [reply webpack.config.js:5]: I didn't understand why running webpack used 2 configurations. especially if I am just trying to do `webpack-dev-server --env dev` (which is what gets called with `npm start`) If you have other suggestions to get this working locally I am open to them. this is just the path of least change I found

File: config/webpack-prod.config.js (hunk 2 above)
Comment (maniax89) [reply config/webpack-prod.config.js:15]: webpack was complaining:

```
/Users/user/Dev/react-querybuilder/node_modules/webpack/lib/webpack.js:16
        throw new WebpackOptionsValidationError(webpackOptionsValidationErrors);
        ^
WebpackOptionsValidationError: Invalid configuration object. Webpack has been initialised using a configuration object that does not match the API schema.
 - configuration[0].resolve.extensions[0] should not be empty.
 - configuration[1].output.library should be one of these:
   string | [string]
   If set, export the bundle as library. `output.library` is the name.
 - configuration[1].resolve.extensions[0] should not be empty.
```

**Pull request merged**